node/src/test_utils/mod.rs

```diff
@@ -569,6 +569,44 @@ pub mod unshared_test_utils {
         pub assertions: Box<dyn FnOnce(&mut A) + Send>,
     }
 
+    pub fn capture_numbers_with_separators_from_str(
```

Comment:
Your use of term digit is better. I suggest you do the same thing with the name:

```
capture_digits_with_separators_from_str
```

Also, change the `digit_grouped_by` to simply `length`.